src/java.base/share/classes/sun/security/provider/certpath/RevocationChecker.java

```diff
@@ -83,23 +83,50 @@
         String ocspSubject;
         String ocspIssuer;
         String ocspSerial;
+        boolean ocspNonce;
     }
+    private RevocationProperties rp;
+    private static final int DEFAULT_NONCE_BYTES = 16;
 
     RevocationChecker() {
+        rp = getRevocationProperties();
         legacy = false;
+        setDefaultNonce();
     }
 
     RevocationChecker(TrustAnchor anchor, ValidatorParams params)
         throws CertPathValidatorException
     {
+        rp = getRevocationProperties();
         legacy = true;
+        setDefaultNonce();
         init(anchor, params);
     }
 
+    private void setDefaultNonce() {
+        byte[] nonce = null;
+
+        // Set the nonce by default in OCSP request extension when the sytem property
+        // jdk.security.certpath.ocspNonce=true.
+        if (rp.ocspNonce) {
+            try {
+                setOcspExtensions(List.of(new OCSPNonceExtension(DEFAULT_NONCE_BYTES)));
```

Comment:
I don't think we should use the `PKIXRevocationChecker.setOcspExtensions()` API to add an OCSP Nonce extension. That API is intended to be called by applications. I think the Nonce extension should be set by the implementation only and not exposed via the standard API. Also, a nonce should be unique for each OCSP request, but setting it here means that it could re-use the same nonce for different OCSP requests.

I think a better place to create/add the OCSPExtension is in the `checkOCSP` method, and the extension should be created each time that method is called (if the system property is enabled), so a new nonce is created for each OCSP request.

Reply:
Creating the nonce is moved to checkOCSP() method.